### ISSUE TITLE
[test] add sqlancer test

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#1768]
**Summary**
Add sqlancer test:
https://github.com/sqlancer/sqlancer

There are 2 parts here:
1. build sqlancer test docker image
2. run it in the CI

**Comment by PsiACE** [comment]
I will evaluate it.

/assignme

**Comment by PsiACE** [comment]
Ref: https://github.com/sqlancer/sqlancer/pull/429

**Comment by PsiACE** [comment]
I think we still need to do that, but I probably won't put more effort into it.

Maybe @leiysky will have some suggestions for that.

**Comment by leiysky** [comment]
> I think we still need to do that, but I probably won't put more effort into it.
> 
> Maybe @leiysky will have some suggestions for that.

It's helpful but not suitable for regression test in CI. We may have a periodic job for it one day.

**Comment by Xuanwo** [comment]
Any updates on this? @PsiACE

**Comment by PsiACE** [comment]
> Any updates on this? @PsiACE
This is an OSPP 2022 project and is assigned to @hanyisong  . You can view the progress at https://github.com/datafuse-extras/sqlancer/pull/1 .

Now that we have almost supported sqlancer, there is still one important issue(#7463) left to resolve.

**Comment by BohuTANG** [comment]
cc @Xuanwo and @leiysky for the help on issue #7463 , it's blocking the sqlancer project.

**Comment by leiysky** [comment]
> cc @Xuanwo and @leiysky for the help on issue #7463 , it's blocking the sqlancer project.

I'm already working on this. It would be fixed as soon as #6885 is finised.

**Comment by PsiACE** [comment]
https://github.com/sqlancer/sqlancer/pull/568 has been merged and I think we have a implementation of the sqlancer test. The next step is to fix the known issues and explore sqlancer. So I think we can close this issue.

Also, with the upstream merge, I have merged https://github.com/datafuse-extras/sqlancer/pull/1 as well. https://github.com/datafuse-extras/sqlancer is the target repository for OSPP2022 assessment, so let's keep it until the end of the event.

Thank you @hanyisong  for your excellent work in making Databend more reliable.
